nodejs/week3/src/backend/api/meals.js

```diff
@@ -0,0 +1,219 @@
+    .then((result) => response.json(result))
+    .catch((ex) => {
+      response.status(400).send("Bad request").end();
+      console.log(ex);
+    });
+});
+
+//getting meals by id
+
+router.get("/:id", async (req, res) => {
+  const mealId = parseInt(req.params.id);
+  if (!mealId) res.status(400).send(`Bad request, id should be a number.`);
+  else {
+    const meal = await knex("meal").where({ id: mealId });
+    if (meal.length === 0)
+      res.status(404).send(`Not found, there is no meal with id: ${mealId}.`);
+    else res.json(meal);
+  }
+});
+
+const editMeal = async ({ body, id }) => {
+  const {
+    title,
+    description,
+    location,
+    created_date,
+    max_reservation,
+    price,
+  } = body;
+  const meal = await knex.from("meal").select("*").where({
+    id: id,
+  });
+  if (meal.length === 0) {
+    throw new HttpError("Bad request", `Contact not found: ID ${id}!`, 404);
+  }
+  const updateMeals = {
+    title: title,
+    description: description,
+    price: price,
+  };
+  if (Object.keys(updateMeals).length !== 0) {
+    return await knex("meal")
+      .where({
+        id: id,
+      })
+      .update(updateMeals);
+  } else return "Nothing updated!";
+};
+
+router.put("/:id", async (req, res) => {
+  editMeal({
+    body: req.body,
+    id: req.params.id,
+  })
+    .then((result) => res.json(result))
+    .catch((error) => {
+      res.status(400).send("Bad request").end();
+      console.log(error);
+    });
+});
+
+router.delete("/", async (req, res) => {
+  deleteMeal({
+    body: req.body,
+  })
+    .then((result) => res.json(result))
+    .catch((error) => {
+      res.status(400).send("Bad request").end();
+      console.log(error);
+    });
+});
+
+const deleteMeal = async ({ body }) => {
+  try {
+    if (!body.id) {
+      throw new HttpError("Bad request", "Id not found", 400);
+    }
+    return knex("meal")
+      .where({
+        id: body.id,
+      })
+      .del();
+  } catch (err) {
+    console.log(err);
+    return "something went wrong, try again";
+  }
+};
+
+//GET api/meals/ query parameters
+
+const getByLimit = async (limit) => {
+  try {
+    return await knex("meal").select("*").orderBy("id", "DESC").limit(limit);
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const getByMaxPrice = async () => {
+  try {
+    return await knex("meal")
+      .select("*")
+      .from("meal")
+      .where("price", "<", "maxPrice");
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const getByCreatedAfter = async () => {
+  try {
+    return await knex("meal")
+      .select("*")
+      .from("meal")
+      .where("created_date", ">", "createdAfter");
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const title = async () => {
+  try {
+    return await knex("meal")
+      .select("*")
+      .from("meal")
+      .where("title", "like", "`%${title}%`");
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+router.get("/", async (request, response) => {
```

Comment:
I recommend not to create new get router for limit, maxPrice, etc., because since you already have one `router.get('/' ` in the beginning of this file, that is going to be used, and therefore all meals are going to be returned instead of getting to the filters. 
So try to add these different filters into that first function (starts on line 46).